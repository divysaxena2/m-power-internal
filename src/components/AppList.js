import React, { Component } from 'react';
import getTheme from '../../native-base-theme/components'
import material from '../../native-base-theme/variables/material';
import { StyleSheet, View } from 'react-native';
import { StyleProvider, List, ListItem, Left, Right, Body, Text } from 'native-base';


type Props = {};
export default class AppList extends Component<Props> {

    render() {
        return (
            <StyleProvider style={getTheme(material)}>
                <List>
                    <ListItem style={styles.listItem}>
                        <View style={{ paddingLeft: 15 }}>
                            <Text style={styles.date}>5 May,2019</Text>
                        </View>
                        <Body style={{ paddingLeft: 20 }}>
                            <Text style={styles.provider}>Kaiser Permanente</Text>
                            <Text note style={{ fontSize: 11 }}>Bill</Text>
                        </Body>
                        <View style={{ paddingLeft: 33 }}>
                            <Text style={styles.amount}>$500</Text>
                            <Text style={styles.payText}>Pay Now</Text>
                        </View>
                    </ListItem>
                </List>
            </StyleProvider>)
    }
}
const styles = StyleSheet.create({
    listItem: {
        height: 56,
        width: 363
    },
    date: {
        height: 16,
        width: 83.3,
        color: '#000000',
        fontSize: 12,
        letterSpacing: 0.47,
        lineHeight: 16
    },
    amount: {
        height: 20,
        width: 60,
        color: '#000000',
        //fontFamily: "Roboto Medium"; 
        fontSize: 15,
        letterSpacing: 1.06,
        lineHeight: 20,
        textAlign: 'right'

    },
    provider: {
        height: 16,
        width: 128,
        color: '#000000',
        fontSize: 12,
        letterSpacing: 0.47,
        lineHeight: 16
    },
    payText: {
        height: 15,
        opacity: 0.4,
        color: '#000000',
        // fontFamily: Roboto;
        fontSize: 11,
        letterSpacing: 1.02,
        lineHeight: 15,
        textAlign: 'right',
        textDecorationLine: 'underline'
    }
});