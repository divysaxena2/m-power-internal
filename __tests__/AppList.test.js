import 'react-native';
import React from 'react';
import AppList from '../src/components/AppList';

import renderer from 'react-test-renderer';

it('renders correctly', () => {
    const tree = renderer.create(
        <AppList />
    ).toJSON();
    expect(tree).toMatchSnapshot();
});