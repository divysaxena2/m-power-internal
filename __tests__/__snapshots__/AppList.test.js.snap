// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  closeOnRowBeginSwipe={false}
  closeOnRowPress={true}
  closeOnScroll={true}
  directionalDistanceChangeThreshold={2}
  disableLeftSwipe={false}
  disableRightSwipe={false}
  leftOpenValue={0}
  previewFirstRow={false}
  recalculateHiddenLayout={false}
  rightOpenValue={0}
  style={Object {}}
  swipeToOpenPercent={50}
>
  <View
    accessible={true}
    isTVSelectable={true}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={null}
  >
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "backgroundColor": "transparent",
            "borderBottomWidth": 0.5,
            "borderColor": "#fff",
            "flexDirection": "row",
            "marginLeft": 18,
            "paddingBottom": 15,
            "paddingRight": 18,
            "paddingTop": 15,
          },
          Object {
            "height": 56,
            "width": 363,
          },
        ]
      }
    >
      <View
        style={
          Object {
            "paddingLeft": 15,
          }
        }
      >
        <Text
          style={
            Array [
              Object {
                "alignSelf": "center",
                "color": "#000",
                "fontFamily": "Roboto",
                "fontSize": 16,
              },
              Object {
                "color": "#000000",
                "fontSize": 12,
                "height": 16,
                "letterSpacing": 0.47,
                "lineHeight": 16,
                "width": 83.3,
              },
            ]
          }
          uppercase={false}
        >
          5 May,2019
        </Text>
      </View>
      <View
        style={
          Array [
            Object {
              "alignItems": null,
              "alignSelf": null,
              "flex": 1,
            },
            Object {
              "paddingLeft": 20,
            },
          ]
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "#000",
                "fontFamily": "Roboto",
                "fontSize": 16,
                "marginLeft": 12,
                "marginRight": 12,
              },
              Object {
                "color": "#000000",
                "fontSize": 12,
                "height": 16,
                "letterSpacing": 0.47,
                "lineHeight": 16,
                "width": 128,
              },
            ]
          }
          uppercase={false}
        >
          Kaiser Permanente
        </Text>
        <Text
          note={true}
          style={
            Array [
              Object {
                "color": "#808080",
                "fontFamily": "Roboto",
                "fontSize": 14,
                "fontWeight": "200",
                "marginLeft": 12,
                "marginRight": 12,
              },
              Object {
                "fontSize": 11,
              },
            ]
          }
          uppercase={false}
        >
          Bill
        </Text>
      </View>
      <View
        style={
          Object {
            "paddingLeft": 33,
          }
        }
      >
        <Text
          style={
            Array [
              Object {
                "alignSelf": "center",
                "color": "#000",
                "fontFamily": "Roboto",
                "fontSize": 16,
              },
              Object {
                "color": "#000000",
                "fontSize": 15,
                "height": 20,
                "letterSpacing": 1.06,
                "lineHeight": 20,
                "textAlign": "right",
                "width": 60,
              },
            ]
          }
          uppercase={false}
        >
          $500
        </Text>
        <Text
          style={
            Array [
              Object {
                "alignSelf": "center",
                "color": "#000",
                "fontFamily": "Roboto",
                "fontSize": 16,
              },
              Object {
                "color": "#000000",
                "fontSize": 11,
                "height": 15,
                "letterSpacing": 1.02,
                "lineHeight": 15,
                "opacity": 0.4,
                "textAlign": "right",
                "textDecorationLine": "underline",
              },
            ]
          }
          uppercase={false}
        >
          Pay Now
        </Text>
      </View>
    </View>
  </View>
</View>
`;
